opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -794,43 +798,160 @@ static int mca_btl_base_am_rdma_progress (void)
     }));
 }
 
+static int mca_btl_base_am_atomic_lock_64 (int64_t *operand, opal_atomic_int64_t *addr,
+                                           mca_btl_base_atomic_op_t op, opal_mutex_t *lock) {
+
+    int64_t result = 0;
+
+    if(lock) {
+        OPAL_THREAD_LOCK(lock);
+    }
+
+    switch (op) {
+        case MCA_BTL_ATOMIC_ADD:
+            result = *addr;
+            *addr += *operand;
+            break;
+        case MCA_BTL_ATOMIC_AND:
+            result = *addr;
+            *addr &= *operand;
+            break;
+        case MCA_BTL_ATOMIC_OR:
+            result = *addr;
+            *addr |= *operand;
+            break;
+        case MCA_BTL_ATOMIC_XOR:
+            result = *addr;
+            *addr ^= *operand;
+            break;
+        case MCA_BTL_ATOMIC_SWAP:
+            result = *addr;
+            *addr  = *operand;
+            break;
+        case MCA_BTL_ATOMIC_MIN:
+            result = *addr;
+            if(result > *operand) {
+                *addr = *operand;
+            }
+            break;
+        case MCA_BTL_ATOMIC_MAX:
+            result = *addr;
+            if(result < *operand) {
+                *addr = *operand;
+            }
+            break;
+        default: {
+            if(lock) {
+                OPAL_THREAD_UNLOCK(lock);
+            }
+            return OPAL_ERR_BAD_PARAM;
+        }
+    }
+
+    if(lock) {
+        OPAL_THREAD_UNLOCK(lock);
+    }
+
+    *operand = result;
+    return OPAL_SUCCESS;
+
+}
+
 static int mca_btl_base_am_atomic_64 (int64_t *operand, opal_atomic_int64_t *addr,
                                       mca_btl_base_atomic_op_t op)
 {
     int64_t result = 0;
 
     switch (op) {
-    case MCA_BTL_ATOMIC_ADD:
-        result = opal_atomic_fetch_add_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_AND:
-        result = opal_atomic_fetch_and_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_OR:
-        result = opal_atomic_fetch_or_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_XOR:
-        result = opal_atomic_fetch_xor_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_SWAP:
-        result = opal_atomic_swap_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_MIN:
-        result = opal_atomic_fetch_min_64 (addr, *operand);
-        break;
-    case MCA_BTL_ATOMIC_MAX:
-        result = opal_atomic_fetch_max_64 (addr, *operand);
-        break;
-    default:
-        return OPAL_ERR_BAD_PARAM;
+        case MCA_BTL_ATOMIC_ADD:
```

Comment:
I randomly picked two places where `switch` statements are used in opal and both do not indent the `case` labels. These white-space changes seem unnecessary.

Reply:
My own preference sneaking in again. Fixed

Reply:
good that we now have the clang-format to keep all this in a consistent state ;)